fastfuels_sdk/exports.py

```diff
@@ -313,6 +326,336 @@ def export_zarr_to_fds(zroot: zarr.hierarchy.Group,
         fout.write(template.substitute(fds_attrs))
 
+def calibrate_duet(zroot: zarr.hierarchy.Group,
+                   output_dir: Path | str,
+                   param_dir: Path | str = None,
+                   keep_duet: str = None,
+                   **kwargs: float) -> None:
+    """
+    Calibrate the values of the surface bulk density output from DUET by changing
+    the range or center and spread. Calibration can be based off of SB40-derived 
+    fuel loading values from Landfire, or from user inputs of summary statistics
+    of bulk density.
+
+    Possible summary statistics:
+    - Mean and standard deviation OR
+    - Maximum and minimum
+
+    These inputs may be supplied for the following fuel types:
+    - Grass AND/OR
+    - Litter (currently combined deciduous and coniferous) 
+    OR
+    - Total
+
+    Inputs are provided in the keyword arguments. For any inputs that are not provided,
+    calibration is based off of min and max values of SB40-derived fuel loads. No user
+    inputs are necessary.
+
+    The fuelgrid zarr is required to parameterize Landfire queries, with the following
+    groups and arrays:
+
+    Required zarr groups:
+    - surface
+
+    Required zarr arrays:
+    - surface/bulk-density
+
+    Parameters
+    ----------
+    zroot : zarr.hierarchy.Group
+        The root group of the zarr file.
+    output_dir : Path | str
+        The output directory where DUET files are stored, 
+        and where the calibrated file will be written to.
+    param_dir : Path | str = None
+        The directory containing the Sb40 parameters table, if
+        different from output_dir. Defaults to None
+    keep_duet : str
+        Which fuel type should not be modified by the calibration, 
+        if any. One of "grass" or "litter"
+    **kwargs : float
+        Keyword arguments are reserved for summary statistics
+        of target values for fuel loading. Calibration is based 
+        on either a mean and standard deviation OR a maximum 
+        and minimum. Valid fuel types are "grass", "litter", 
+        or "total". Arguements should be given in the format 
+        fueltype_statistic = float (eg: litter_mean = 0.8).
+
+    Returns
+    -------
+    None
+        Calibrated array of DUET surface bulk density is saved to the output directory.
+    
+    TODO: make keep_duet boolean 
+    """
+    # Validate the zarr file
+    required_groups = ["surface"]
+    required_arrays = {
+        "surface": ["bulk-density"]
+    }
+    _validate_zarr_file(zroot, required_groups, required_arrays)
+    
+    # Convert the output and SB40 parameters directories to a Path objects if they are strings
+    if isinstance(output_dir, str):
+        output_dir = Path(output_dir)
+    if isinstance(param_dir, str):
+        param_dir = Path(param_dir)
+
+    # Get fuel type(s) from kwargs
+    valid_ftypes = [["grass"],["litter"],["total"],["grass","litter"],["litter","grass"],["none"]]
+    fuel_types = []
```

Comment:
I think we could replace this with the oneline:

fuel_types = [ftype for ftype in ["total", "grass", "litter"] if any(key.startswith(ftype) for key in kwargs)] or ["none"]